### PR TITLE
supply: supply pack datums, some reworks

### PR DESCRIPTION
**Pull request by silicons**
tl;dr refactors supplypacks to a new system

- all access locks yanked from crates other than the ones that actually need to be locked
- preps system for reworks, translates stuff over to a new format
- adds unified subsystem parsing & support for other descriptors for cargo crate contents other than typepath

### CODE REVIEW

File: code/datums/armor/station/armor.dm
Comment (ktoma36) [inline code/datums/armor/station/armor.dm:1]: :)

File: code/modules/client/client_procs.dm
Comment (ktoma36) [inline code/modules/client/client_procs.dm:1]: Odd pr to have this in but fair

File: code/modules/clothing/spacesuits/void/station.dm [hunk 1]
```diff
@@ -1,242 +1,3 @@
-// Station voidsuits
-//Engineering
-/obj/item/clothing/head/helmet/space/void/engineering
```
Comment (ktoma36) [inline code/modules/clothing/spacesuits/void/station.dm:3]: Wow I hate this file path

**Review by deathride58 — APPROVED** [review]
im not readin all that but im happy you fixed the code or im sad you broke the thing

**Review by ktoma36 — APPROVED** [review]

**Pull request merged**